src/Jobs/UpdateMerchantPriceBenchmarks.php

```diff
@@ -0,0 +1,111 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Jobs;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\ActionScheduler\ActionSchedulerInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter\MerchantCenterService;
+use Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter\PriceBenchmarks;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Google\MerchantPriceBenchmarks;
+use Throwable;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class UpdateMerchantPriceBenchmarks
+ *
+ * Update Product Stats
+ *
+ * Note: The job will not start if it is already running or if the Google Merchant Center account is not connected.
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Jobs
+ *
+ * @since 2.6.4
+ */
+class UpdateMerchantPriceBenchmarks extends AbstractActionSchedulerJob {
+	/**
+	 * @var MerchantCenterService
+	 */
+	protected $merchant_center;
+
+	/**
+	 * @var PriceBenchmarks
+	 */
+	protected $price_benchmarks;
+
+	/**
+	 * UpdateMerchantPriceBenchmarks constructor.
+	 *
+	 * @param ActionSchedulerInterface  $action_scheduler
+	 * @param ActionSchedulerJobMonitor $monitor
+	 * @param MerchantCenterService     $merchant_center
+	 * @param PriceBenchmarks           $price_benchmarks
+	 */
+	public function __construct(
+		ActionSchedulerInterface $action_scheduler,
+		ActionSchedulerJobMonitor $monitor,
+		MerchantCenterService $merchant_center,
+		PriceBenchmarks $price_benchmarks
+	) {
+		parent::__construct( $action_scheduler, $monitor );
+		$this->merchant_center  = $merchant_center;
+		$this->price_benchmarks = $price_benchmarks;
+	}
+
+	/**
+	 * Get the name of the job.
+	 *
+	 * @return string
+	 */
+	public function get_name(): string {
+		return 'update_merchant_price_benchmarks';
+	}
+
+	/**
+	 * Can the job be scheduled.
+	 *
+	 * @param array|null $args
+	 *
+	 * @return bool Returns true if the job can be scheduled.
+	 */
+	public function can_schedule( $args = [] ): bool {
+		return parent::can_schedule( $args ) && $this->merchant_center->is_connected();
+	}
+
+	/**
+	 * Process the job.
+	 *
+	 * @param int[] $items An array of job arguments.
+	 *
+	 * @throws JobException If the merchant product statuses cannot be retrieved..
+	 */
+	public function process_items( array $items ) {
```

Comment:
@joemcgill As we don't have pagination how we use the `$items` here?

Reply:
That’s exactly what I initially considered during implementation.  

At the moment, the API calls `PriceBenchmarks::update_price_benchmarks`, which handles cleaning up the DB table before inserting the new benchmark data.  

If we add pagination arguments to `get_benchmark_data`, we’ll need to rethink how we handle the DB cleanup and data insertion since cleaning up the table before each paginated fetch would wipe out previously inserted data. 🤔

Can we follow up on this later once we have a clearer understanding on how the API data is structured and delivered especially if it supports pagination or incremental updates?